contrib_bots/lib/giphy/giphy.py

```diff
@@ -0,0 +1,140 @@
+# To use this plugin, you need to set up the Giphy API key for this bot in
+# ~/.giphy_config
+
+from __future__ import absolute_import
+from __future__ import print_function
+import six
+import requests
+import logging
+import sys
+import os
+import re
+if six.PY2:
+    from six.moves.configparser import SafeConfigParser
+else:
+    from configparser import SafeConfigParser
+
+GIPHY_TRANSLATE_API = 'http://api.giphy.com/v1/gifs/translate'
+
+if not os.path.exists(os.environ['HOME'] + '/.giphy_config'):
```

Comment:
This is a tricky thing.  You can't really put logic like this in __main__, because we need the library to fail quickly if you run it from `run.py`.  We will probably eventually have `run.py` call an `on_startup` function where you could verify that this config exists.

I don't think this should hold up this commit.